doc/adr/0010-reducing-snapshot-size.md

```diff
@@ -0,0 +1,409 @@
+# 10. Reducing global snapshot size
+
+Date: 2022-05-19
+
+## Status
+
+Proposed
+
+## Context
+
+Currently, the `GlobalSnapshotInfo` type represents an aggregated view of:
+
+- balance,
+- last transaction reference,
+- last state channel snapshot hash
+
+for every address ever used with a ledger.
+
+Addresses that were not used with a ledger have default values implicitly assumed: 
+- `Balance.empty`,
+- `TransactionReference.empty`,
+- `Hash.empty`.
+
+The problem is that every single `GlobalSnapshot` contains all this data for all addresses ever used. This is suboptimal
+because majority of this data never changes between two consecutive snapshots, and it causes an increasing-only size of
+snapshots. In the long run this can cause issues with storage, network and memory.
+
+## Decision
+
+Replacing existing `GlobalSnapshotInfo` type with a type `GlobalSnapshotInfos` which is a product type of:
+1. `SnapshotInfo` - Contains the same fields as the `GlobalSnapshotInfo` type and represents overrides to the 
+    `balances`, `lastTransactionRef` and `lastStateChannelSnapshotHashes` that are applied by this snapshot. 
+    Missing values for an address should be treated as lack of overrides - the value from previous snapshot still holds.
```

Comment:
The cool thing is that it will be transparent for OpenSearch. We used the same pattern for balances where we just sort by `ordinal` and we return last available balance when user asks for balance on specific `ordinal`. I like the idea.